Sources/Crashes/CrashCollection.swift

```diff
@@ -80,8 +87,29 @@ public final class CrashCollection {
             didFindCrashReports(pixelParameters, processedData) {
                 Task {
                     for payload in processedData {
-                        await self.crashSender.send(payload)
+                        let crcid = self.crashCollectionStorage.object(forKey: Const.crcidKey) as? String
+                        let result =  await self.crashSender.send(payload, crcid: crcid)
+
+                        switch result.result {
+                        case .success:
+                            Logger.general.debug("Crash Collection - Sending Crash Report: succeeded")
+                            if let receivedCRCID = result.response?.allHeaderFields[CrashReportSender.httpHeaderCRCID] as? String {
+                                if crcid != receivedCRCID {
```

Comment:
why not just always set the received crcid, regardless of whether it matches or not? Are there performance implications?

Reply:
It’s unlikely to be an issue either way, but in my experience it’s best practice not to re-store an existing value, because it can trigger code observing a value when the value hasn’t actually changed.